internal/api/mfa.go

```diff
@@ -205,6 +367,128 @@ func (a *API) ChallengeFactor(w http.ResponseWriter, r *http.Request) error {
 	})
 }
 
+func (a *API) verifySMSFactor(w http.ResponseWriter, r *http.Request, user *models.User, factor *models.Factor, params *VerifyFactorParams) error {
+	ctx := r.Context()
+	config := a.config
+	db := a.db.WithContext(ctx)
+	if !config.MFA.SMS.VerifyEnabled {
+		return unprocessableEntityError(ErrorCodeMFASMSVerifyDisabled, "MFA verification is disabled for sms factors")
+	}
+	currentIP := utilities.GetIPAddress(r)
+
+	if !factor.IsOwnedBy(user) {
+		return internalServerError(InvalidFactorOwnerErrorMessage)
+	}
```

Comment:
hmm i don't think we should return an internal server error here - maybe forbidden error makes more sense? cc @hf

Reply:
Will change the codes for MFA (Phone) -  we can decide to what to do with existing status code for checked in code (TOTP) later